src/components/Header.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { UpdateTodo } from '../types/UpdateTodo';
+
+interface Props {
+  todos: Todo[];
+  handleAddingNewTodo: (title: string) => void;
+  isLoading: boolean;
+  isAllTodosCompleted: boolean;
+  handleUpdatingTodo: (todoId: number, args: UpdateTodo) => void;
+}
+
+export const Header: React.FC<Props> = ({
+  todos,
+  handleAddingNewTodo,
+  isLoading,
+  isAllTodosCompleted,
+  handleUpdatingTodo,
+}) => {
+  const [newTodo, setNewTodo] = useState<string>('');
+
+  const inputField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputField.current) {
+      inputField.current.focus();
+    }
+  }, []);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    handleAddingNewTodo(newTodo);
+    setNewTodo('');
+  };
+
+  const handleResetingCompletedTodos = () => {
+    todos.map(todo => {
+      if (todo.completed) {
+        handleUpdatingTodo(todo.id, { completed: !todo.completed });
+      }
+
+      return null;
+    });
+  };
```

Comment:
That functionality must change completed status to both way. 
"Complete" when one of todos is not complete.
"!Complete" when all of todos is complete.